client/my-sites/plugins/plugin-automated-transfer/index.jsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Notice from 'components/notice';
+import NoticeAction from 'components/notice/notice-action';
+
+class PluginAutomatedTransfer extends Component {
+
+	static propTypes = {
+		translate: PropTypes.func,
+		plugin: PropTypes.object,
+	}
+
+	static defaultProps = {
+		plugin: {
+			name: 'Plugin',
+		},
+	}
+
+	state = {
+		transferStep: 'start',
+	}
+
+	setNoticeText = ( pluginName = '' ) => {
+		const { translate } = this.props;
+		const { transferStep } = this.state;
+
+		switch ( transferStep ) {
+			case 'start': return translate( 'Installing %s…', {
+				args: pluginName,
+				comment: 'Installing plugin…',
+			} );
+			case 'setup': return translate( 'Now configuring your site. This may take a few minutes.' );
+			case 'leaving': return translate( "Don't leave quite yet! Just a bit longer." );
+			case 'conflicts': return translate( 'Sorry, we found some conflicts to fix before proceding.' );
+			case 'complete': return translate( 'Successfully installed %s!', {
+				args: pluginName,
+				context: 'Successfully installed plugin!',
+			} );
+		}
+	}
+
+	setStatus = transferStep => {
+		switch ( transferStep ) {
+			case 'conflicts': return 'is-error';
+			case 'complete': return 'is-success';
+			default: return 'is-info';
+		}
+	}
+
+	setIcon = transferStep => {
+		switch ( transferStep ) {
+			case 'conflicts': return 'notice';
+			case 'complete': return 'checkmark';
+			default: return 'sync';
+		}
+	}
+
+	testChangeState = () => {
+		const { transferStep } = this.state;
+		switch ( transferStep ) {
+			case 'start': this.setState( { transferStep: 'setup' } ); break;
+			case 'setup': this.setState( { transferStep: 'leaving' } ); break;
+			case 'leaving': this.setState( { transferStep: 'conflicts' } ); break;
+			case 'conflicts': this.setState( { transferStep: 'complete' } ); break;
+			case 'complete': this.setState( { transferStep: 'start' } ); break;
+		}
+	}
```

Comment:
I don't suppose that this is appropriate yet, but hopefully soon or maybe before this PR ends. if this were controlled via app state we could use the console dispatcher to change states without needing to add the test code specifically in the module.